client/app/components/Users.jsx

```diff
@@ -2,38 +2,27 @@ import React from 'react';
 import { Link } from 'react-router';
 import AjaxPromise from 'ajax-promise';
 import Store from '../reducers/store.js';
+import loadingUntil from '../reducers/loading.js';
 import { Card, CardTitle, CardText, RaisedButton, TextField } from 'material-ui';
 
 var Users = React.createClass({
 
   componentDidMount: function() {
     if (!this.props.users) {
-      // TODO: More robust way of determining loading state.
-      Store.dispatch({
-        type: "LOADING",
-        isLoading: true
-      });
-
-      AjaxPromise
+      let fetchUsers = AjaxPromise
         .get('/api/user')
         .then(function (response) {
           console.log("load users", response);
           Store.dispatch({
             type: "LOAD_USERS",
             users: response
           });
-          Store.dispatch({
-            type: "LOADING",
-            isLoading: false
-          });
         })
         .catch(function(err){
           console.log("/api/user/index error", err);
-          Store.dispatch({
-            type: "LOADING",
-            isLoading: false
-          });
-        })
+        });
+
+      loadingUntil(fetchUsers);
```

Comment:
@dannyvassallo now this is all you have to do to handle loading state in the store. just pass a promise to `loadingUntil` and the store will be `isLoading` until that promise and all others have been resolved. 

It also solves another potential issue which would arise if you were to dispatch `LOADING` from multiple origins. Now there is only one source of truth for loading promises, and `isLoading` will never be out of sync; and that is `reducers/loading.js`